server/common/app_config.py

```diff
@@ -19,74 +38,312 @@ def todict(self):
+                    context["messsagefn"]("Warning: --annotations-file ignored as annotations are disabled.")
+                if dirname is not None:
+                    context["messagefn"]("Warning: --annotations-dir ignored as annotations are disabled.")
+
+            if self.annotations__ontology__enable:
+                context["messagefn"](
+                    "Warning: --experimental-annotations-ontology" " ignored as annotations are disabled."
+                )
+            if self.annotations__ontology__obo_location is not None:
+                context["messagefn"](
+                    "Warning: --experimental-annotations-ontology-obo" " ignored as annotations are disabled."
+                )
+
+    def handle_embeddings(self, context):
+        self.__check_attr("embeddings__names", (list, tuple))
+        self.__check_attr("embeddings__enable_reembedding", bool)
+
+        if self.single_dataset__datapath:
+            if self.embeddings__enable_reembedding:
+                matrix_data_loader = MatrixDataLoader(self.single_dataset__datapath)
+                if matrix_data_loader.matrix_data_type() != MatrixDataType.H5AD:
+                    raise ConfigurationError("'enable-reembedding is only supported with H5AD files.")
+                if self.anndata_adaptor__backed:
+                    raise ConfigurationError("enable-reembedding is not supported when run in --backed mode.")
+
+    def handle_diffexp(self, context):
+        self.__check_attr("diffexp__enable", bool)
+        self.__check_attr("diffexp__lfc_cutoff", float)
+
+        if self.single_dataset__datapath:
+            with context["matrix_cache"].data_adaptor(self.single_dataset__datapath, self) as data_adaptor:
+                if self.diffexp__enable and data_adaptor.parameters.get("diffexp_may_be_slow", False):
+                    context["messagefn"](
+                        f"CAUTION: due to the size of your dataset, "
+                        f"running differential expression may take longer or fail."
+                    )
+
+    def handle_cxg_adaptor(self, context):
+        self.__check_attr("cxg_adaptor__tiledb_ctx", dict)
+        from server.data_cxg.cxg_adaptor import CxgAdaptor
+        CxgAdaptor.set_tiledb_context(self.cxg_adaptor__tiledb_ctx)
+
+    def handle_anndata_adaptor(self, context):
+        self.__check_attr("anndata_adaptor__backed", bool)
 
     def get_title(self, data_adaptor):
-        return self.title if self.title else data_adaptor.get_title()
+        return self.single_dataset__title if self.single_dataset__title else data_adaptor.get_title()
 
     def get_about(self, data_adaptor):
-        return self.about if self.about else data_adaptor.get_about()
+        return self.single_dataset__about if self.single_dataset__about else data_adaptor.get_about()
 
-    def get_config(self, data_adaptor, annotation=None):
+    def get_client_config(self, data_adaptor, annotation=None):
```

Comment:
I suggested adding a comment describing the purpose of this method, eg, 

"""
Return the configuration as required by the /config REST route
"""

Reply:
Will do.